### PR TITLE
app_service_environment: add field `resource_group_name`

### PR DESCRIPTION
**Pull request by magodo**
This PR allows user to explicitly specify the resource group for app_service_environment, instead of implicitly using the one from Subnet.
In order to keep compatibility, the newly added `resource_group_name` is set as `Optional`+`Computed`, which should be changed to `Required` when a major version of provider bumps.

Fix /terraform-providers/terraform-provider-azurerm#6699.

### CODE REVIEW

File: azurerm/helpers/azure/resource_group.go [hunk 1]
```diff
@@ -35,6 +35,16 @@ func SchemaResourceGroupNameForDataSource() *schema.Schema {
 	}
 }
 
+func SchemaResourceGroupNameOC() *schema.Schema {
```
[the remark this answers is not in the document]
Comment (magodo) [reply azurerm/helpers/azure/resource_group.go:38]: Sure

File: azurerm/internal/services/web/app_service_environment_resource.go [hunk 2]
```diff
@@ -86,18 +86,16 @@ func resourceArmAppServiceEnvironment() *schema.Resource {
 				}, false),
 			},
 
+			// TODO: Make it "Required" in next major release
```
[the remark this answers is not in the document]
Comment (magodo) [reply azurerm/internal/services/web/app_service_environment_resource.go]: Good idea!

**Comment by magodo** [comment]
@jackofallops Thank you for your comment!
I have updated the code accordingly, PTAL!

**Comment by jackofallops** [comment]
Tests pass (2 failures unrelated to PR / transient)
image

**Pull request merged**